contracts/types/LockData.sol

```diff
@@ -0,0 +1,123 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity ^0.8.20;
+
+/// @notice The LockData holds global information for the lock data structure: the length and nonzeroDeltaCount.
+/// @dev The left most 128 bits is the length, or total number of active lockers.
+/// @dev The right most 128 bits is the nonzeroDeltaCount, or total number of nonzero deltas over all active + completed lockers.
+/// @dev Located in transient storage.
+type LockData is uint256;
+
+using LockDataLibrary for LockData global;
+
+function toLockData(uint128 _length, uint128 _nonzeroDeltaCount) pure returns (LockData lockData) {
```

Comment:
Given its going to be transient storage, should we consider just putting these in different slots? Every function that writes to this slot, only updates 1 of the 2 values - they never _both_ get updated. It feels like it would be easier to just separate them?
`push` - only touches length
`pull` - only touches length
`incrementNonzeroDeltaCount` - only touches delta count
`decrementNonzeroDeltaCount` - only touches delta count

Imo it might make more sense to have
```
NON_ZERO_DELTA_COUNT_SLOT = x
LOCKERS_SLOT = y
```

where
```
sload(LOCKERS_SLOT) => length
sload(LOCKERS_SLOT+1) => first locker
```
this is more like how actual arrays are encoded in storage

Reply:
interesting, yeah we could do that and compare gas. tload/tstore still 100 gas so its not crazy to pack though